--- a/src/cmd/cgo/gcc.go
+++ b/src/cmd/cgo/gcc.go
@@ -2004,7 +2004,7 @@ func (p *Package) gccErrors(stdin []byte, extraArgs ...string) string {
 	// trailing "-" at the end.
 	li := len(nargs) - 1
 	last := nargs[li]
-	nargs[li] = "-O0"
+	nargs[li] = "-O1"
 	nargs = append(nargs, extraArgs...)
 	nargs = append(nargs, last)
 
--- a/src/runtime/testdata/testprogcgo/sigpanic.go
+++ b/src/runtime/testdata/testprogcgo/sigpanic.go
@@ -8,7 +8,7 @@ package main
 // We want to test unwinding from sigpanic into C code (without a C symbolizer).
 
 /*
-#cgo CFLAGS: -O0
+#cgo CFLAGS: -O1
 
 char *pnil;
 
--- a/src/runtime/testdata/testprogcgo/traceback.go
+++ b/src/runtime/testdata/testprogcgo/traceback.go
@@ -9,7 +9,7 @@ package main
 // We use a fake traceback, and a symbolizer that dumps a string we recognize.
 
 /*
-#cgo CFLAGS: -g -O0
+#cgo CFLAGS: -g -O1
 
 // Defined in traceback_c.c.
 extern int crashInGo;
