diff --git a/sysdeps/i386/configure b/sysdeps/i386/configure
index 5b55c5affe..c63470b2be 100644
--- a/sysdeps/i386/configure
+++ b/sysdeps/i386/configure
@@ -27,7 +27,7 @@ cat > conftest.c <<EOF
 int _start (void) { int a, b, c; __sync_val_compare_and_swap (&a, b, c); return 0; }
 EOF
 if ! { ac_try='${CC-cc} $CFLAGS $CPPFLAGS $LDFLAGS
-		     -O0 -nostdlib -nostartfiles
+		     -O0 -mindirect-branch=keep -nostdlib -nostartfiles
 		     -S conftest.c -o - | fgrep "__sync_val_compare_and_swap"
 		     1>&5'
   { { eval echo "\"\$as_me\":${as_lineno-$LINENO}: \"$ac_try\""; } >&5
diff --git a/sysdeps/i386/configure.ac b/sysdeps/i386/configure.ac
index 19ef33f34a..dd90daef8d 100644
--- a/sysdeps/i386/configure.ac
+++ b/sysdeps/i386/configure.ac
@@ -23,7 +23,7 @@ fi
 LIBC_COMPILER_BUILTIN_INLINED(
   [__sync_val_compare_and_swap],
   [int a, b, c; __sync_val_compare_and_swap (&a, b, c);],
-  [-O0],
+  [-O0 -mindirect-branch=keep],
   [libc_cv_unsupported_i386=no],
   [AC_MSG_ERROR([
 *** Building with -march=i386/-mcpu=i386 is not supported.
