diff -ur a/engine/build.sh b/engine/build.sh
--- a/engine/build.sh	2019-08-14 12:03:50.000000000 +0000
+++ b/engine/build.sh	2020-01-16 04:50:29.980642534 +0000
@@ -185,7 +185,7 @@
         *)
         B2_CXX="${CXX} -x c++ -std=c++11"
         B2_CXXFLAGS_RELEASE="-O2 -s"
-        B2_CXXFLAGS_DEBUG="-O0 -g"
+        B2_CXXFLAGS_DEBUG="-O1 -g"
         esac
     ;;
 
@@ -236,7 +236,7 @@
         fi
         B2_CXX="${CXX} -xc++"
         B2_CXXFLAGS_RELEASE="-O3 -s"
-        B2_CXXFLAGS_DEBUG="-O0 -g -p"
+        B2_CXXFLAGS_DEBUG="-O1 -g -p"
     ;;
 
     vacpp)
@@ -324,7 +324,7 @@
         B2_CXX="${CXX} -x c++ -O3 -std=c++11"
         B2_TOOLSET=clang
         B2_CXXFLAGS_RELEASE="-O3 -s"
-        B2_CXXFLAGS_DEBUG="-O0 -fno-inline -g"
+        B2_CXXFLAGS_DEBUG="-O1 -Wno-inline -g"
     ;;
 
     tru64cxx)
diff -ur a/tools/gcc.jam b/tools/gcc.jam
--- a/tools/gcc.jam	2019-08-14 12:03:50.000000000 +0000
+++ b/tools/gcc.jam	2020-01-16 04:50:55.767796283 +0000
@@ -704,11 +704,11 @@
 ###
 
 # Declare flags and action for compilation.
-toolset.flags gcc.compile OPTIONS <optimization>off   : -O0 ;
+toolset.flags gcc.compile OPTIONS <optimization>off   : -O1 ;
 toolset.flags gcc.compile OPTIONS <optimization>speed : -O3 ;
 toolset.flags gcc.compile OPTIONS <optimization>space : -Os ;
 
-toolset.flags gcc.compile OPTIONS <inlining>off  : -fno-inline ;
+toolset.flags gcc.compile OPTIONS <inlining>off  : -Wno-inline ;
 toolset.flags gcc.compile OPTIONS <inlining>on   : -Wno-inline ;
 toolset.flags gcc.compile OPTIONS <inlining>full : -finline-functions -Wno-inline ;
 
diff -ur a/tools/gcc.py b/tools/gcc.py
--- a/tools/gcc.py	2019-08-14 12:03:50.000000000 +0000
+++ b/tools/gcc.py	2020-01-16 04:50:43.605723773 +0000
@@ -278,11 +278,11 @@
 flags('gcc.compile', 'PCH_FILE', ['<pch>on'], ['<pch-file>'])
 
 # Declare flags and action for compilation
-flags('gcc.compile', 'OPTIONS', ['<optimization>off'], ['-O0'])
+flags('gcc.compile', 'OPTIONS', ['<optimization>off'], ['-O1'])
 flags('gcc.compile', 'OPTIONS', ['<optimization>speed'], ['-O3'])
 flags('gcc.compile', 'OPTIONS', ['<optimization>space'], ['-Os'])
 
-flags('gcc.compile', 'OPTIONS', ['<inlining>off'], ['-fno-inline'])
+flags('gcc.compile', 'OPTIONS', ['<inlining>off'], ['-Wno-inline'])
 flags('gcc.compile', 'OPTIONS', ['<inlining>on'], ['-Wno-inline'])
 flags('gcc.compile', 'OPTIONS', ['<inlining>full'], ['-finline-functions', '-Wno-inline'])
 
