diff --git a/base/unix-gcc.mak b/base/unix-gcc.mak
index 121e9c6..ad6f983 100644
--- a/base/unix-gcc.mak
+++ b/base/unix-gcc.mak
@@ -366,7 +366,7 @@ GCFLAGS=  -Wall -Wstrict-prototypes -Wundef -Wmissing-declarations -Wmissing-pro
 # and shared object builds.
 
 CFLAGS_STANDARD= -O2 -DNDEBUG
-CFLAGS_DEBUG= -g -O0
+CFLAGS_DEBUG= -g -O1
 CFLAGS_PROFILE=-pg  -O2 -DNDEBUG
 CFLAGS_SO=-fPIC
 
diff --git a/configure.ac b/configure.ac
index e4feaea..741a8b5 100644
--- a/configure.ac
+++ b/configure.ac
@@ -272,21 +272,21 @@ case $host in
         *-linux*|*-gnu)
         if test x"$GCC" = x"yes"; then
             CC_OPT_FLAGS_TO_TRY="-O2 -DNDEBUG"
-            CC_DBG_FLAGS_TO_TRY="-gdwarf-2 -g3 -O0"
+            CC_DBG_FLAGS_TO_TRY="-gdwarf-2 -g3 -O1"
             CC_VISIBILITY_FLAGS_TO_TRY="-fvisibility=hidden"
         fi
         ;;
         *bsd*)
         if test x"$GCC" = x"yes"; then
             CC_OPT_FLAGS_TO_TRY="-O2 -DNDEBUG"
-            CC_DBG_FLAGS_TO_TRY="-gdwarf-2 -g3 -O0"
+            CC_DBG_FLAGS_TO_TRY="-gdwarf-2 -g3 -O1"
             CC_VISIBILITY_FLAGS_TO_TRY="-fvisibility=hidden"
         fi
         ;;
         *-darwin*)
         if test x"$GCC" = x"yes"; then
             CC_OPT_FLAGS_TO_TRY="-O2 -DNDEBUG"
-            CC_DBG_FLAGS_TO_TRY="-gdwarf-2 -g3 -O0"
+            CC_DBG_FLAGS_TO_TRY="-gdwarf-2 -g3 -O1"
             CC_VISIBILITY_FLAGS_TO_TRY="-fvisibility=hidden"
         fi
         SET_DT_SONAME=""
@@ -294,7 +294,7 @@ case $host in
         *-mingw*|*-msys*)
         if test x"$GCC" = x"yes"; then
             CC_OPT_FLAGS_TO_TRY="-O2 -DNDEBUG"
-            CC_DBG_FLAGS_TO_TRY="-gdwarf-2 -g3 -O0"
+            CC_DBG_FLAGS_TO_TRY="-gdwarf-2 -g3 -O1"
             CC_VISIBILITY_FLAGS_TO_TRY="-fvisibility=hidden"
         fi
         SET_DT_SONAME=""
@@ -308,16 +308,16 @@ case $host in
         fi
         if test x"$GCC" = x"yes"; then
             CC_OPT_FLAGS_TO_TRY="-O2 -DNDEBUG"
-            CC_DBG_FLAGS_TO_TRY="-gdwarf-2 -g3 -O0"
+            CC_DBG_FLAGS_TO_TRY="-gdwarf-2 -g3 -O1"
             CC_VISIBILITY_FLAGS_TO_TRY="-fvisibility=hidden"
         else
-            CC_DBG_FLAGS_TO_TRY="-g -O0"
+            CC_DBG_FLAGS_TO_TRY="-g -O1"
         fi
         ;;
         *-aix*)
         if test x"$GCC" = x"yes"; then
             CC_OPT_FLAGS_TO_TRY="-O2 -DNDEBUG"
-            CC_DBG_FLAGS_TO_TRY="-gdwarf-2 -g3 -O0"
+            CC_DBG_FLAGS_TO_TRY="-gdwarf-2 -g3 -O1"
             SET_DT_SONAME="so"
             CC_VISIBILITY_FLAGS_TO_TRY="-fvisibility=hidden"
         fi
