diff -ur ghostscript-9.52.orig/base/unix-gcc.mak ghostscript-9.52/base/unix-gcc.mak
--- ghostscript-9.52.orig/base/unix-gcc.mak	2020-10-02 12:31:24.682144014 +0000
+++ ghostscript-9.52/base/unix-gcc.mak	2020-10-02 12:40:50.611905295 +0000
@@ -377,7 +377,7 @@
 # and shared object builds.
 
 CFLAGS_STANDARD= -O2 -DNDEBUG
-CFLAGS_DEBUG= -g -O0
+CFLAGS_DEBUG= -g -O1
 CFLAGS_PROFILE=-pg  -O2 -DNDEBUG
 CFLAGS_SO=-fPIC
 
diff -ur ghostscript-9.52.orig/configure.ac ghostscript-9.52/configure.ac
--- ghostscript-9.52.orig/configure.ac	2020-10-02 12:31:24.322141620 +0000
+++ ghostscript-9.52/configure.ac	2020-10-02 12:42:23.961523266 +0000
@@ -286,26 +286,26 @@
         *-linux*|*-gnu)
         if test $ac_cv_prog_gcc = yes; then
             CC_OPT_FLAGS_TO_TRY="-O2 -DNDEBUG"
-            CC_DBG_FLAGS_TO_TRY="-gdwarf-2 -g3 -O0"
+            CC_DBG_FLAGS_TO_TRY="-gdwarf-2 -g3 -O1"
         fi
         ;;
         *bsd*)
         if test $ac_cv_prog_gcc = yes; then
             CC_OPT_FLAGS_TO_TRY="-O2 -DNDEBUG"
-            CC_DBG_FLAGS_TO_TRY="-gdwarf-2 -g3 -O0"
+            CC_DBG_FLAGS_TO_TRY="-gdwarf-2 -g3 -O1"
         fi
         ;;
         *-darwin*)
         if test $ac_cv_prog_gcc = yes; then
             CC_OPT_FLAGS_TO_TRY="-O2 -DNDEBUG"
-            CC_DBG_FLAGS_TO_TRY="-gdwarf-2 -g3 -O0"
+            CC_DBG_FLAGS_TO_TRY="-gdwarf-2 -g3 -O1"
         fi
         SET_DT_SONAME=""
         ;;
         *-mingw*|*-msys*|*-cygwin*)
         if test $ac_cv_prog_gcc = yes; then
             CC_OPT_FLAGS_TO_TRY="-O2 -DNDEBUG"
-            CC_DBG_FLAGS_TO_TRY="-gdwarf-2 -g3 -O0"
+            CC_DBG_FLAGS_TO_TRY="-gdwarf-2 -g3 -O1"
         fi
         SET_DT_SONAME=""
         ;;
@@ -318,15 +318,15 @@
         fi
         if test $ac_cv_prog_gcc = yes; then
             CC_OPT_FLAGS_TO_TRY="-O2 -DNDEBUG"
-            CC_DBG_FLAGS_TO_TRY="-gdwarf-2 -g3 -O0"
+            CC_DBG_FLAGS_TO_TRY="-gdwarf-2 -g3 -O1"
         else
-            CC_DBG_FLAGS_TO_TRY="-g -O0"
+            CC_DBG_FLAGS_TO_TRY="-g -O1"
         fi
         ;;
         *-aix*)
         if test $ac_cv_prog_gcc = yes; then
             CC_OPT_FLAGS_TO_TRY="-O2 -DNDEBUG"
-            CC_DBG_FLAGS_TO_TRY="-gdwarf-2 -g3 -O0"
+            CC_DBG_FLAGS_TO_TRY="-gdwarf-2 -g3 -O1"
             SET_DT_SONAME="so"
         fi
         ;;
