diff -ur ghostscript-9.53.3.orig/base/unix-gcc.mak ghostscript-9.53.3/base/unix-gcc.mak
--- ghostscript-9.53.3.orig/base/unix-gcc.mak	2020-10-01 08:11:07.000000000 +0000
+++ ghostscript-9.53.3/base/unix-gcc.mak	2020-12-29 16:17:02.586780251 +0000
@@ -377,7 +377,7 @@
 # and shared object builds.
 
 CFLAGS_STANDARD= -O2 -DNDEBUG
-CFLAGS_DEBUG= -g -O0
+CFLAGS_DEBUG= -g -O1
 CFLAGS_PROFILE=-pg  -O2 -DNDEBUG
 CFLAGS_SO=-fPIC
 
diff -ur ghostscript-9.53.3.orig/configure.ac ghostscript-9.53.3/configure.ac
--- ghostscript-9.53.3.orig/configure.ac	2020-10-01 08:11:07.000000000 +0000
+++ ghostscript-9.53.3/configure.ac	2020-12-29 16:16:40.979698582 +0000
@@ -287,21 +287,21 @@
         *-linux*|*-gnu)
         if test $GCC = yes; then
             CC_OPT_FLAGS_TO_TRY="-O2 -DNDEBUG"
-            CC_DBG_FLAGS_TO_TRY="-gdwarf-2 -g3 -O0"
+            CC_DBG_FLAGS_TO_TRY="-gdwarf-2 -g3 -O1"
             CC_VISIBILITY_FLAGS_TO_TRY="-fvisibility=hidden"
         fi
         ;;
         *bsd*)
         if test $GCC = yes; then
             CC_OPT_FLAGS_TO_TRY="-O2 -DNDEBUG"
-            CC_DBG_FLAGS_TO_TRY="-gdwarf-2 -g3 -O0"
+            CC_DBG_FLAGS_TO_TRY="-gdwarf-2 -g3 -O1"
             CC_VISIBILITY_FLAGS_TO_TRY="-fvisibility=hidden"
         fi
         ;;
         *-darwin*)
         if test $GCC = yes; then
             CC_OPT_FLAGS_TO_TRY="-O2 -DNDEBUG"
-            CC_DBG_FLAGS_TO_TRY="-gdwarf-2 -g3 -O0"
+            CC_DBG_FLAGS_TO_TRY="-gdwarf-2 -g3 -O1"
             CC_VISIBILITY_FLAGS_TO_TRY="-fvisibility=hidden"
         fi
         SET_DT_SONAME=""
@@ -309,7 +309,7 @@
         *-mingw*|*-msys*)
         if test $GCC = yes; then
             CC_OPT_FLAGS_TO_TRY="-O2 -DNDEBUG"
-            CC_DBG_FLAGS_TO_TRY="-gdwarf-2 -g3 -O0"
+            CC_DBG_FLAGS_TO_TRY="-gdwarf-2 -g3 -O1"
             CC_VISIBILITY_FLAGS_TO_TRY="-fvisibility=hidden"
         fi
         SET_DT_SONAME=""
@@ -323,16 +323,16 @@
         fi
         if test $GCC = yes; then
             CC_OPT_FLAGS_TO_TRY="-O2 -DNDEBUG"
-            CC_DBG_FLAGS_TO_TRY="-gdwarf-2 -g3 -O0"
+            CC_DBG_FLAGS_TO_TRY="-gdwarf-2 -g3 -O1"
             CC_VISIBILITY_FLAGS_TO_TRY="-fvisibility=hidden"
         else
-            CC_DBG_FLAGS_TO_TRY="-g -O0"
+            CC_DBG_FLAGS_TO_TRY="-g -O1"
         fi
         ;;
         *-aix*)
         if test $GCC = yes; then
             CC_OPT_FLAGS_TO_TRY="-O2 -DNDEBUG"
-            CC_DBG_FLAGS_TO_TRY="-gdwarf-2 -g3 -O0"
+            CC_DBG_FLAGS_TO_TRY="-gdwarf-2 -g3 -O1"
             SET_DT_SONAME="so"
             CC_VISIBILITY_FLAGS_TO_TRY="-fvisibility=hidden"
         fi
