diff -ur libgpg-error-1.27.orig/src/Makefile.am libgpg-error-1.27/src/Makefile.am
--- libgpg-error-1.27.orig/src/Makefile.am	2017-02-28 12:33:38.000000000 +0000
+++ libgpg-error-1.27/src/Makefile.am	2018-03-02 17:30:14.805020382 +0000
@@ -271,7 +271,7 @@
 
 
 mkheader: mkheader.c Makefile
-	$(CC_FOR_BUILD) -g -O0 -I. -I$(srcdir) -o $@ $(srcdir)/mkheader.c
+	$(CC_FOR_BUILD) -g -I. -I$(srcdir) -o $@ $(srcdir)/mkheader.c
 
 parts_of_gpg_error_h = 	 	\
 	gpg-error.h.in  	\
diff -ur libgpg-error-1.27.orig/src/Makefile.in libgpg-error-1.27/src/Makefile.in
--- libgpg-error-1.27.orig/src/Makefile.in	2017-02-28 14:03:41.000000000 +0000
+++ libgpg-error-1.27/src/Makefile.in	2018-03-02 17:32:30.681811735 +0000
@@ -1326,7 +1326,7 @@
 		$(srcdir)/errnos.in >$@
 
 mkheader: mkheader.c Makefile
-	$(CC_FOR_BUILD) -g -O0 -I. -I$(srcdir) -o $@ $(srcdir)/mkheader.c
+	$(CC_FOR_BUILD) -g -I. -I$(srcdir) -o $@ $(srcdir)/mkheader.c
 
 @FORCE_USE_SYSCFG_FALSE@./lock-obj-pub.native.h: Makefile gen-posix-lock-obj$(EXEEXT) posix-lock-obj.h
 @FORCE_USE_SYSCFG_FALSE@	./gen-posix-lock-obj >$@
