diff --git a/meson/mesonbuild/compilers/mixins/clike.py b/meson/mesonbuild/compilers/mixins/clike.py
index 09ad837b1..215749e1d 100644
--- a/meson/mesonbuild/compilers/mixins/clike.py
+++ b/meson/mesonbuild/compilers/mixins/clike.py
@@ -185,7 +185,7 @@ def get_compile_only_args(self) -> T.List[str]:
         return ['-c']
 
     def get_no_optimization_args(self) -> T.List[str]:
-        return ['-O0']
+        return ['-O1']
 
     def get_output_args(self, target: str) -> T.List[str]:
         return ['-o', target]
diff --git a/meson/run_unittests.py b/meson/run_unittests.py
index 1c0e6f620..d6e2cc312 100755
--- a/meson/run_unittests.py
+++ b/meson/run_unittests.py
@@ -7296,7 +7296,7 @@ def test_compiler_check_flags_order(self):
             self.assertEqual(cmd[1], '-I/tmp')
             # Verify that -O3 set via the environment is overridden by -O0
             Oargs = [arg for arg in cmd if arg.startswith('-O')]
-            self.assertEqual(Oargs, [Oflag, '-O0'])
+            self.assertEqual(Oargs, [Oflag, '-O1'])
 
     def _test_stds_impl(self, testdir: str, compiler: 'Compiler') -> None:
         has_cpp17 = (compiler.get_id() not in {'clang', 'gcc'} or
