diff -ur bind-9.11.2-P1.orig/lib/isc/Makefile.in bind-9.11.2-P1/lib/isc/Makefile.in
--- bind-9.11.2-P1.orig/lib/isc/Makefile.in	2018-01-04 05:28:11.000000000 +0000
+++ bind-9.11.2-P1/lib/isc/Makefile.in	2018-03-19 15:52:12.782420601 +0000
@@ -93,7 +93,7 @@
 @BIND9_MAKE_RULES@
 
 safe.@O@: safe.c
-	${LIBTOOL_MODE_COMPILE} ${CC} ${ALL_CFLAGS} @CCNOOPT@ \
+	${LIBTOOL_MODE_COMPILE} ${CC} ${ALL_CFLAGS} \
 		-c ${srcdir}/safe.c
 
 version.@O@: version.c
diff -ur bind-9.11.2-P1.orig/lib/isc/safe.c bind-9.11.2-P1/lib/isc/safe.c
--- bind-9.11.2-P1.orig/lib/isc/safe.c	2018-01-04 05:28:11.000000000 +0000
+++ bind-9.11.2-P1/lib/isc/safe.c	2018-03-19 15:51:06.457122911 +0000
@@ -19,10 +19,10 @@
 
 isc_boolean_t
 isc_safe_memequal(const void *s1, const void *s2, size_t n) {
-	isc_uint8_t acc = 0;
+	volatile isc_uint8_t acc = 0;
 
 	if (n != 0U) {
-		const isc_uint8_t *p1 = s1, *p2 = s2;
+		volatile const isc_uint8_t *p1 = s1, *p2 = s2;
 
 		do {
 			acc |= *p1++ ^ *p2++;
@@ -34,9 +34,9 @@
 
 int
 isc_safe_memcompare(const void *b1, const void *b2, size_t len) {
-	const unsigned char *p1 = b1, *p2 = b2;
+	volatile const unsigned char *p1 = b1, *p2 = b2;
 	size_t i;
-	int res = 0, done = 0;
+	volatile int res = 0, done = 0;
 
 	for (i = 0; i < len; i++) {
 		/* lt is -1 if p1[i] < p2[i]; else 0. */
