--- a/mesonbuild/compilers/c.py
+++ b/mesonbuild/compilers/c.py
@@ -635,7 +635,7 @@ class Xc16CCompiler(Xc16Compiler, CCompiler):
         return []
 
     def get_no_optimization_args(self) -> T.List[str]:
-        return ['-O0']
+        return ['-O1']
 
     def get_output_args(self, target: str) -> T.List[str]:
         return ['-o%s' % target]
--- a/mesonbuild/compilers/mixins/clike.py
+++ b/mesonbuild/compilers/mixins/clike.py
@@ -182,7 +182,7 @@ class CLikeCompiler(Compiler):
         return ['-c']
 
     def get_no_optimization_args(self) -> T.List[str]:
-        return ['-O0']
+        return ['-O1']
 
     def get_output_args(self, target: str) -> T.List[str]:
         return ['-o', target]
--- a/run_unittests.py
+++ b/run_unittests.py
@@ -6294,7 +6294,7 @@ class LinuxlikeTests(BasePlatformTests):
             self.assertEqual(cmd[1], '-I/tmp')
             # Verify that -O3 set via the environment is overridden by -O0
             Oargs = [arg for arg in cmd if arg.startswith('-O')]
-            self.assertEqual(Oargs, [Oflag, '-O0'])
+            self.assertEqual(Oargs, [Oflag, '-O1'])
 
     def _test_stds_impl(self, testdir, compiler, p: str):
         has_cpp17 = (compiler.get_id() not in {'clang', 'gcc'} or
