diff -ur meson-0.51.2.orig/mesonbuild/compilers/c.py meson-0.51.2/mesonbuild/compilers/c.py
--- meson-0.51.2.orig/mesonbuild/compilers/c.py	2019-08-25 19:17:02.000000000 +0000
+++ meson-0.51.2/mesonbuild/compilers/c.py	2020-01-14 19:12:59.966210516 +0000
@@ -357,9 +357,6 @@
     def get_compile_only_args(self):
         return []
 
-    def get_no_optimization_args(self):
-        return ['-optimize=0']
-
     def get_output_args(self, target):
         return ['-output=obj=%s' % target]
 
diff -ur meson-0.51.2.orig/mesonbuild/compilers/clike.py meson-0.51.2/mesonbuild/compilers/clike.py
--- meson-0.51.2.orig/mesonbuild/compilers/clike.py	2019-08-25 19:17:02.000000000 +0000
+++ meson-0.51.2/mesonbuild/compilers/clike.py	2020-01-14 19:14:24.117679565 +0000
@@ -123,7 +123,7 @@
         return ['-c']
 
     def get_no_optimization_args(self):
-        return ['-O0']
+        return ['-O1']
 
     def get_compiler_check_args(self):
         '''
diff -ur meson-0.51.2.orig/run_unittests.py meson-0.51.2/run_unittests.py
--- meson-0.51.2.orig/run_unittests.py	2019-08-26 16:28:03.000000000 +0000
+++ meson-0.51.2/run_unittests.py	2020-01-14 19:12:47.390140456 +0000
@@ -4486,9 +4486,9 @@
             # Verify that -I flags from the `args` kwarg are first
             # This is set in the '40 has function' test case
             self.assertEqual(cmd[1], '-I/tmp')
-            # Verify that -O3 set via the environment is overridden by -O0
+            # Verify that -O3 set via the environment is overridden by -O1
             Oargs = [arg for arg in cmd if arg.startswith('-O')]
-            self.assertEqual(Oargs, [Oflag, '-O0'])
+            self.assertEqual(Oargs, [Oflag, '-O1'])
 
     def _test_stds_impl(self, testdir, compiler, p):
         lang_std = p + '_std'
