diff -ur meson-0.55.3.orig/mesonbuild/compilers/c.py meson-0.55.3/mesonbuild/compilers/c.py
--- meson-0.55.3.orig/mesonbuild/compilers/c.py	2020-09-11 18:35:34.000000000 +0000
+++ meson-0.55.3/mesonbuild/compilers/c.py	2020-10-19 13:05:00.368382078 +0000
@@ -501,7 +501,7 @@
         return []
 
     def get_no_optimization_args(self):
-        return ['-O0']
+        return ['-O1']
 
     def get_output_args(self, target):
         return ['-o%s' % target]
diff -ur meson-0.55.3.orig/mesonbuild/compilers/mixins/clike.py meson-0.55.3/mesonbuild/compilers/mixins/clike.py
--- meson-0.55.3.orig/mesonbuild/compilers/mixins/clike.py	2020-09-10 16:39:24.000000000 +0000
+++ meson-0.55.3/mesonbuild/compilers/mixins/clike.py	2020-10-19 13:05:44.362659565 +0000
@@ -169,7 +169,7 @@
         return ['-c']
 
     def get_no_optimization_args(self):
-        return ['-O0']
+        return ['-O1']
 
     def get_compiler_check_args(self):
         '''
diff -ur meson-0.55.3.orig/run_unittests.py meson-0.55.3/run_unittests.py
--- meson-0.55.3.orig/run_unittests.py	2020-10-19 13:04:02.049013713 +0000
+++ meson-0.55.3/run_unittests.py	2020-10-19 13:06:55.325106436 +0000
@@ -6124,7 +6124,7 @@
             self.assertEqual(cmd[1], '-I/tmp')
             # Verify that -O3 set via the environment is overridden by -O0
             Oargs = [arg for arg in cmd if arg.startswith('-O')]
-            self.assertEqual(Oargs, [Oflag, '-O0'])
+            self.assertEqual(Oargs, [Oflag, '-O1'])
 
     def _test_stds_impl(self, testdir, compiler, p: str):
         has_cpp17 = (compiler.get_id() not in {'clang', 'gcc'} or
